workspaces/redhat-resource-optimization/packages/backend/src/index.ts

```diff
@@ -15,6 +15,79 @@
  */
 
 import { createBackend } from '@backstage/backend-defaults';
+import { createBackendModule } from '@backstage/backend-plugin-api';
```

Comment:
It seems you took the hard and painful path here with the auth provider...

PTAL at [this patch](https://gitkraken.dev/link/drafts/2fd0cd98-5604-46fe-8d45-9f68bf63d1f3?type=patch). My understanding is that you created this entire custom auth provider because sometimes the email was not available. 
Didn't any of [these resolvers](https://backstage.io/docs/auth/github/provider/#resolvers) provide a good solution?

Reply:
And the app-config should be something like this
```yaml
auth:
  environment: development
  providers:
    github:
      development:
        # y0n1
        clientId: '...'
        clientSecret: '...'
        # jkilzi
        # clientId: '...'
        # clientSecret: '...'
        signIn:
          resolvers:
            # See https://backstage.io/docs/auth/github/provider#resolvers for more resolvers
            - resolver: usernameMatchingUserEntityName
    guest:
      dangerouslyAllowOutsideDevelopment: true
```

Reply:
@jkilzi I had tried the above way but it was not working as we needed the user in catalog and for that I had to write a custom provider.